packages/go/lab/loader.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2025 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package lab
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/specterops/bloodhound/dawgs/graph"
+)
+
+// GraphFixture is the JSON representation of the graph we are importing.
+type GraphFixture struct {
+	Nodes         []Node `json:"nodes"`
+	Relationships []Edge `json:"relationships"`
+}
+
+// Node is the JSON representation of a graph node.
+type Node struct {
+	// ID is the local identifier for the Node within the file.
+	// This ID is not preserved when imported into the database.
+	ID string `json:"id"`
+
+	// Caption will be used as the `name` property of a Node.
+	Caption string `json:"caption"`
+
+	// Labels are the node types. This is equivalent to what we call
+	// Kinds in BloodHound.
+	Labels []string `json:"labels"`
+
+	// Properties is the key:value map used for storing extra information
+	// on the Node. For timestamp properties, you can use a timestamp function
+	// like `NOW()` or `NOW()-3600` where `-3600` modifies the timestamp to
+	// be 3600 seconds in the past.
+	Properties map[string]string `json:"properties"`
+}
+
+// Edge is the JSON representation of a graph edge.
+type Edge struct {
+	// FromID is the local Node identifier for the start of an Edge.
+	FromID string `json:"fromId"`
+
+	// ToID is the local Node identifier for the end of an Edge.
+	ToID string `json:"toId"`
+
+	// Type is the 'label' we apply to the Edge. This is synonymous to
+	// the edge Kind in BloodHound.
+	Type string `json:"type"`
+
+	// Properties is the key:value map used for storing extra information
+	// on the Edge.
+	Properties map[string]string `json:"properties"`
+}
+
+// LoadGraphFixture requires a graph.Database interface and a GraphFixture struct.
+// It will import the nodes and edges from the GraphFixture and inserts them into
+// the graph database. It uses the `id` property of the nodes as the local
+// identifier and then maps them to database IDs, meaning that the `id` given in
+// the file will not be preserved.
+func LoadGraphFixture(db graph.Database, g *GraphFixture) error {
```

Comment:
I still feel like we only need to expose one of these functions. For now, we have no need for LoadGraphFixture outside of the package, so I'd prefer to keep it internal

Reply:
i'm realizing i did actually collapse one of the functions, but i think i do want to keep these two separated. i believe in the future that this will be simplified once generic ingest is introduced, but if the concern is public interface, then we should have a separate discussion around that.